main.go

```diff
@@ -190,17 +211,20 @@ func main() {
 		os.Exit(1)
 	}
 
-	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
-		setupLog.Error(err, "unable to set up health check")
-		os.Exit(1)
-	}
-	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
-		setupLog.Error(err, "unable to set up ready check")
-		os.Exit(1)
-	}
+	leaderElectionContext, cancelLeaderElection := context.WithCancel(context.Background())
+	go func() {
+		setupLog.Info("starting leader election manager")
+		if err := leaderElectionMgr.Start(leaderElectionContext); err != nil {
+			setupLog.Error(err, "Leader Election Manager exited non-zero")
+			os.Exit(1)
```

Comment:
something seems off to me with the use of os.Exit() here and below

maybe just need to use waitgroup to wait for these managers to complete.

what if we os.Exit() before any defer statements are executed. it may not really be an issue IDK.

Reply:
+1

Reply:
TBH, I don't know: if one manager returns an error while the others are running, it should signal a stop and wait. But it could become a little bit complicated and there is a risk of getting deadlock as far as I can see.
I can explore a little more if you think it's worth

Reply:
+1 on looking how to streamline this